src/lib.rs

```diff
@@ -485,25 +538,25 @@ impl BacktracePrinter {
             .frames()
             .iter()
             .flat_map(|frame| frame.symbols())
-            .map(|sym| Frame {
+            .enumerate()
+            .map(|(n, sym)| Frame {
                 name: sym.name().map(|x| x.to_string()),
                 lineno: sym.lineno(),
                 filename: sym.filename().map(|x| x.into()),
+                n,
             })
             .collect();
 
-        // Try to find where the interesting part starts...
-        let top_cutoff = frames
-            .iter()
-            .rposition(Frame::is_post_panic_code)
-            .map(|x| x + 1)
-            .unwrap_or(0);
+        // Turn them into `Frame` objects and print them.
+        let num_frames = frames.len();
 
-        // Try to find where language init frames start ...
-        let bottom_cutoff = frames
-            .iter()
-            .position(Frame::is_runtime_init_code)
-            .unwrap_or_else(|| frames.len());
+        let mut frames = frames.iter().collect();
```

Comment:
Would it be better to use `Vec::retain` here and pass `&mut Vec<Frames>` as the argument to filters? I'm worried that this will cause a lot of unnecessary allocations.

Reply:
Well, would it be better. Performance-wise? Probably. Interface-wise. Hmm.

Reply:
Well, it would even be shorter than calling `.into_iter()` and `.collect()`. 

```rust
fn filter(frames: &mut Vec<Frame>) {
    frames.retain(|x| matches!(&x.name, Some(n) if n.starts_with("oof")));
}
```

Actually, this is elegant AF. Let's go for it. Excellent idea!